### PR TITLE
Form builder page

### PR DESCRIPTION
**Pull request by m-wynn**
This is the base work for the form builder page.  It could still use some more widgets, support for validation, mobile improvements, dragability and a way to actually send things to the back-end.

### CODE REVIEW

File: constellation_forms/static/constellation_forms/css/forms.css [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+.widget-type {
+    color: #aaaaaa;
+    font-size: 20px;
+    font-weight: bold;
+    text-align: right;
+}
+
+.form-extra {
+    background-color: #dddddd;
```
Comment (dinoocch) [inline constellation_forms/static/constellation_forms/css/forms.css]: Could we make this lighter? like `#f2f2f2`?

File: constellation_forms/static/constellation_forms/css/forms.css [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+/* Fast transition for box-shadow */
+.form-part::after {
+    box-shadow: 0 2px 5px 0 rgba(0,0,0,.14),0 1px 10px 0 rgba(0,0,0,.12),0 2px 4px -1px rgba(0,0,0,.2);
+    content: "";
+    height: 100%;
+    margin: -16px;
+    opacity: 0;
+    position: absolute;
```
Comment (dinoocch) [inline constellation_forms/static/constellation_forms/css/forms.css]: The shadow on my title and description form-part is far to the right.
form_title

File: constellation_forms/static/constellation_forms/js/widgets/selectfield.handlebars.html [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        <input class="mdl-textfield__input" type="text" id="selectfield-title-{{ id }}">
+        <label class="mdl-textfield__label" for="selectfield-title-{{ id }}">Field Title</label>
+    </div>
+    <ul id="choices-list-{{ id }}" class="mdl-list mdl-cell--12-col choices-list mdl-cell--order-3">
+        <li class="mdl-list__item">
+            <span class="mdl-list__item-primary-content">
+                <i class="material-icons">check_box</i>
+                <div class="mdl-textfield mdl-js-textfield mdl-cell mdl-cell--4-col">
+                    <input class="mdl-textfield__input" type="text" name="choice-1" id="selectfield-choice-{{ id }}-1">
+                    <label class="mdl-textfield__label" for="selectfield-choice-{{ id }}-1">Choice</label>
```
Comment (dinoocch) [inline constellation_forms/static/constellation_forms/js/widgets/selectfield.handlebars.html:15]: We should also add an option to delete fields eventually...

File: constellation_forms/static/constellation_forms/css/forms.css (hunk 2 above)
Comment (m-wynn) [reply constellation_forms/static/constellation_forms/css/forms.css]: I'm unable to reproduce this.  Are you able to narrow it down to a browser, resolution, or action?  I tried on Firefox-nightly, Chromium, and Opera, both zooming out and stretching the window too.

File: constellation_forms/static/constellation_forms/css/forms.css (hunk 1 above)
Comment (m-wynn) [reply constellation_forms/static/constellation_forms/css/forms.css]: 👍

File: constellation_forms/static/constellation_forms/js/widgets/selectfield.handlebars.html (hunk 3 above)
Comment (m-wynn) [reply constellation_forms/static/constellation_forms/js/widgets/selectfield.handlebars.html:15]: done

File: constellation_forms/static/constellation_forms/css/forms.css (hunk 2 above)
Comment (m-wynn) [reply constellation_forms/static/constellation_forms/css/forms.css]: Firefox stable has funny default left and top values for ::after elements.  Fixed

**Comment by m-wynn** [comment]
```json
{
   "widgets":[
      {
         "type":"textfield",
         "title":"aaa",
         "description":"bbb",
         "required":"on"
      },
      {
         "type":"selectfield",
         "title":"ccc",
         "choice-1":"ddd",
         "choice-2":"eee",
         "choice-3":"fff",
         "required":"on"
      },
      {
         "type":"datefield",
         "title":"ggg",
         "description":"hhh",
         "required":"on"
      }
   ],
   "meta":{
      "title":"abc",
      "description":"def"
   },
   "options":{

   }
}
```

**Review by the-maldridge — APPROVED** [review]
:+1:

**Comment by dinoocch** [comment]
:shipit:

**Pull request merged**